### ISSUE TITLE
Update `publish()` to take links

### ISSUE THREAD

**Issue by kjschmidt913** [MLMI2-CSSI/foundry#187]
Need to take "links" to pass onto the MDF connect_client. To do this one, need to add helper function to MDF connect client first (something like add_links()

- [x] add helper functions to Connect Client
- [ ] add logic in Foundry publish() to pass on links metadata

before this story also included passing in all datacite data. preserving that text here in case it's needed in the future:

need to take all necessary keys to add to the datacite block (called dc) (e.g. description, resource_type, etc)
For "description", in data cite it's just Description, and then you can give it a DescriptionType:“Abstract”. Link to dc metadata schema: https://github.com/materials-data-facility/data-schemas/blob/12bb3c7da2a5d8667e5cdb8fe90f18b77094af4d/schemas/dc.json#L411

So it'd be something like:
```
{
    "dc": {
        "descriptions": [{
            "description": "Blah blah blah",
            "descriptionType": "Abstract"
        }]
    }
}
```
Things I need in the dc block:

```
def create_dc_block(self, title, authors,
                        affiliations=None, publisher=None, publication_year=None,
                        resource_type=None, description=None, dataset_doi=None,
                        related_dois=None, subjects=None,
                        **kwargs):
```

**Comment by kjschmidt913** [comment]
dependent on issue #23 being completed first

**Comment by blaiszik** [comment]
Check Foundry Miro board for metadata format.

Some ideas

```"links": [{
 "type": "dataset_paper",
   "doi": "10.1072/xyz",
 "bibtex": "<bibtex text>",
   "url": "https://doi.org/10.1072/xyz"
  },
"type": "other_paper",
 "doi": "10.1072/xyz",
 "bibtex": "<bibtex text>",
   "url": "https://doi.org/10.1072/xyz"
  },
  {
   "type": "model",
   "doi": "10.1072/thy",
   "bibtex": "<bibtex text>"
   "url": "https://doi.org/10.1072/thy"
  },
  {
   "type": "code",
   "doi": "10.1072/thy",
   "<bibtex text>"
   "url": "https://doi.org/10.1072/thy"
  },
 {
 "type": "dataset",
   "doi": "10.1072/thy",
   "<bibtex text>"
   "url": "https://doi.org/10.1072/thy"
  }
 ]
```

```
(this)dataset_paper: The paper where the dataset was released
other_paper: Other papers associated with or using the dataset
model: Links to models
code: Links to code
```

**Comment by ascourtas** [comment]
may be blocked by an MDF connect client piece?

**Comment by blue442** [comment]
This looks pretty straightforward at this point as we can call the MDFConnectClient [`add_links()`](https://github.com/materials-data-facility/connect_client/blob/ce019467414b7188a8af6183c7226e41f9369f88/mdf_connect_client/mdfcc.py#L421C13-L421C13) method during the call to `Foundry.publish_dataset()` - there are just two questions I had:

1. Should the 'links' argument be passed as an explicit keyword argument to `Foundry.publish_dataset()`, or as one of the **kwargs (and do we have any general guidelines on this)?
2. Do we want to validate the link data structure (e.g. using [this](https://github.com/materials-data-facility/data-schemas/blob/master/schemas/links.json) as a guide)?

**Comment by ascourtas** [comment]
Hi Steve! 
1. There isn't a clearcut guide -- I think we try to limit the explicit keyword argument to only the most necessary things. I think it's ok to add it as a `kwarg` for now -- we may want to refactor how we do args for this function in the future anways
2. I can't tell just by looking at `get_links()` and the MDFConnectClient class if any validation happens (what is `self.mdf`?) but we should indeed validate the link data structure, but probably on the MDF side of things. I would suggest tracking down to see if that happens anywhere, and if not we should open another issue

**Comment by ascourtas** [comment]
Blocked on class restructure #395

**Comment by kjschmidt913** [comment]
Since this is covered in the refactor work and the refactor issue doesn't link to this one, I'm going to close this issue
